### DISCUSSION TITLE
Could I simulate a transaction without costing any gases?

### DISCUSSION THREAD

**Discussion by fenghaojiang** [MystenLabs/sui#7037] [Q&A] [answered]
Could I simulate a transaction without costing any gases? 
I know `Sui` is totally different from `Ethereum`. But I got a question. It seems that the APIs of Sui Node provided can return the objects' info during the runtime. Like `sui_getObject`. 
If I know the object id of the Sui Coin or NFT, I know the balance and the owner. Is that right? I don't have to simulate the transaction or to read the owner or balance by calling the contract fetching the data from EVM runtime like I did in `Ethereum`. I just need to know the object ID and call the `sui_getObect` to know every detail. 
So here is my problem: 
Could I simulate a transaction without costing any gases like in read contract in the `Ethereum`? 
For example: the entry function just returns some information but actually doesn't make some change or just returning the information. WIl the cost of gas be zero or small enough the caller would not have to take it?

**Comment by Dexterbillionz360** [comment, marked as answer]
love your piece.. but gas fee is compulsory if you must simulate transactions.

**Reply by fenghaojiang** [reply]
> love your piece.. but gas fee is compulsory if you must simulate transactions.

So Sui has no `view function`. Would the method `sui_getObject` be enough to fetch data?

**Reply by mztacat** [reply]
why not request gas for that.. are you talking about view only?

**Reply by fenghaojiang** [reply]
> why not request gas for that.. are you talking about view only?

Yes. But all my opinions are from EVM-like blockchains, I think Sui is totally different. I just want to be sure about that.

**Reply by mztacat** [reply]
sure, SUI is based on rust, and i think it should also be able to lazy mint.. but whichever way, you would still need aa bit of gas fam
